torchft/collectives_test.py

```diff
@@ -46,21 +63,22 @@ def _run_parallel_collectives(
 
             self._collect(futures)
 
-        def _run_collective(
+        def _run_all_reduce_collective(
             self,
             pg: ProcessGroup,
-            rank: int,
             device: str,
             tensors_num: int,
             tensor_size: int,
             multiplier: float,
             tolerance: float,
+            reduce_op: ReduceOp,
+            dtype: torch.dtype,
         ) -> None:
             cuda.set_device(device)
             inp = (
                 torch.rand(
                     tensors_num * tensor_size,
-                    dtype=torch.float32,
+                    dtype=dtype,
```

Comment:
curious, if we run with dtypes that aren't supported, is it silently wrong? Or do things blow up